src/components/menu/menu-item/menu-item.component.ts

```diff
@@ -21,14 +21,23 @@ import { MenuItemType } from './menu-item-type.enum';
 export class MenuItemComponent implements OnInit, OnDestroy, FocusableOption {
 
     /** Define if this item is disabled or not */
-    @Input() disabled: boolean = false;
+    @Input() set disabled(value: boolean) {
```

Comment:
Hey Christina, I realise this is just a draft, but just to give you a little bit of insight here, this is a common pattern called coercion, and the Angular CDK offers some utilities to help us.

```typescript
@Input() set disabled(disabled: boolean) {
    this._disabled = coerceBooleanProperty(disabled);
}

get disabled(): boolean {
    return this._disabled;
}
```

This way we don't have to do the ternary in the getter as all conversion is done in the setter. This also allows us to use string values in our html to avoid having to use bindings, e.g.

```html
<button disabled="true"
```

As the `coerceBooleanProperty` function will also convert strings. The problem is that in Angular you can turn on strict type checking, which ensures that you use the correct types in your templates. As `disabled` is expecting a boolean, if I use a string value this will throw an error.

So Angular provides us a way to allow the template to use string or a boolean, but if you are using the disabled property in code we only want to allow boolean.

To do this we add the following property to our component class:

```typescript
static ngAcceptInputType_disabled: boolean | string;
```

This tells Angular to allow a string or boolean type in a template. The CDK also gives us a `BooleanInput` type rather than having to use `boolean | string` each time, but its only in v9 and up so we need to wait until we drop v8 support.

As a side note, you should be able to remove:

```typescript
@HostBinding('attr.disabled') disabledMenu: boolean = null;
```

As you can either add the `HostBinding` to the disabled property as well, or use the host property in the component decorator, e.g.:

```typescript
@Component({
    ....
   host: {
      '[attr.disabled]': 'disabled'
   }
})
```

Which should help simply the code a bit. I believe there is a little bit written about coercion in the Tips and Tricks section, but it could probably be a bit more detailed, so perhaps this might be a good thing to add.

Reply:
Thanks Ashley for the insight it's definitely very helpful and I will do a little more research into it this afternoon and update the tips & tricks page to what I find out

Reply:
Thank you 😊